bugsnag-android-core/src/main/java/com/bugsnag/android/internal/JournaledDocument.kt

```diff
@@ -81,20 +82,40 @@ constructor(
      */
     fun addCommand(command: Journal.Command) {
         synchronized(journal) {
-            if (!isOpen) {
-                throw IllegalStateException("Cannot add commands to a closed document")
-            }
-            try {
-                command.serialize(journalStream)
-            } catch (ex: BufferOverflowException) {
-                snapshot()
-                command.serialize(journalStream)
-            }
-            document = command.apply(document)
-            journal.add(command)
+            requireOpen()
+            internalAddCommand(command)
+        }
+    }
+
+    fun addCommands(vararg commands: Journal.Command) {
```

Comment:
There are ~4 variations on `addCommand/addCommands` doing very similar things - could they be chained so that the important logic is only in one?

Reply:
They can't be chained because they run synchronized code at different levels to keep the amount of CPU churn down depending on how you're adding the commands. The common code has already been moved to `requireOpen()` and `internalAddCommand()`.